### DISCUSSION TITLE
Checking for field value to match defaultValue?

### DISCUSSION THREAD

**Discussion by zoodirector** [TanStack/form#933] [Q&A] [answered]
Is there an API to find out whether the state of a field is matching the value of  `defaultValues` ? I would like to know if the state of the form matches the original. The use case would be to disable a "Save" button in this situation. The value of `isDirty` seems to stay true, even if I change the value of a field back to the original value.

**Comment by Balastrong** [comment, marked as answer]
There's a note on this in [the docs](https://tanstack.com/form/latest/docs/framework/react/guides/basic-concepts#field-state) (right below the image).

You can access `form.options.defaultValues` or `field.options.defaultValue` to have full control on that, like comparing if values are different.

**Reply by zoodirector** [reply]
Okay, so I need to do the comparison myself. I was wondering if there was a convenience property like `form.state.isDefault` which would be true if all fields of the form were equal to their default values.
